src/test/scala/com/microsoft/hyperspace/actions/RefreshActionTest.scala

```diff
@@ -83,16 +94,26 @@ class RefreshActionTest extends SparkFunSuite with SparkInvolvedSuite {
 
   test("validate() passes if old index logs are found with ACTIVE state") {
     testLogEntry.state = ACTIVE
+    updateSourceFiles()
     when(mockLogManager.getLog(anyInt)).thenReturn(Some(testLogEntry))
     val action = new RefreshAction(spark, mockLogManager, mockDataManager)
     action.validate()
   }
 
   test("validate() fails if old index logs found with non-ACTIVE state") {
     testLogEntry.state = CREATING
+    updateSourceFiles()
     when(mockLogManager.getLog(anyInt)).thenReturn(Some(testLogEntry))
     val action = new RefreshAction(spark, mockLogManager, mockDataManager)
     val ex = intercept[HyperspaceException](action.validate())
     assert(ex.getMessage.contains("Refresh is only supported in ACTIVE state"))
   }
+
+  test("validate() fails if there is no source data change.") {
+    testLogEntry.state = ACTIVE
+    when(mockLogManager.getLog(anyInt)).thenReturn(Some(testLogEntry))
+    val action = new RefreshAction(spark, mockLogManager, mockDataManager)
+    val ex = intercept[NoChangesException](action.validate())
+    assert(ex.getMessage.contains("Refresh full aborted"))
```

Comment:
What about checking the full exception msg as it is a fixed/non-parametrized one?

`assert(ex.getMessage.contains("Refresh full aborted as no source data changed"))`